odl/discr/discr_ops.py

```diff
@@ -101,56 +111,353 @@ def _call(self, x, out=None):
+                offset, domain.ndim, param_conv=safe_int_conv)
+            self.__offset = tuple(offset)
+
+            range = _resize_discr(domain, ran_shp, offset, discr_kwargs)
+
+        elif ran_shp is None:
+            if offset is not None:
+                raise ValueError('`offset` can only be combined with '
+                                 '`ran_shp`')
+
+            if not np.allclose(range.cell_sides, domain.cell_sides):
+                raise ValueError(
+                    'cell sides of domain and range differ significantly '
+                    '(difference {})'
+                    ''.format(range.cell_sides - domain.cell_sides))
+
+            self.__offset = self._offset_from_spaces(domain, range)
+
+        else:
+            raise ValueError('cannot combine `range` with `ran_shape`')
+
+        pad_mode = kwargs.pop('pad_mode', 'constant')
+        self.__pad_mode = str(pad_mode).lower()
+        if self.pad_mode not in _SUPPORTED_RESIZE_PAD_MODES:
+            raise ValueError("`pad_mode` '{}' not understood".format(pad_mode))
+
+        self.__pad_const = float(kwargs.pop('pad_const', 0.0))
+
+        # padding mode 'constant' with `pad_const != 0` is not linear
+        linear = (self.pad_mode != 'constant' or self.pad_const == 0.0)
+
+        super().__init__(domain, range, linear=linear)
+
+    @property
+    def offset(self):
+        """Number of cells added to/removed from the left."""
+        return self.__offset
+
+    @property
+    def pad_mode(self):
+        """Padding mode used by this operator."""
+        return self.__pad_mode
+
+    @property
+    def pad_const(self):
+        """Constant used by this operator in case of constant padding."""
+        return self.__pad_const
+
+    @staticmethod
+    def _offset_from_spaces(dom, ran):
+        """Return index offset corresponding to given spaces."""
+        diff_l = np.abs(ran.grid.min() - dom.grid.min())
+        offset_float = diff_l / dom.cell_sides
+        offset = np.around(offset_float).astype(int)
+        if not np.allclose(offset, offset_float):
+            raise ValueError('range is shifted relative to domain by a '
+                             'non-multiple of cell_sides')
+        return tuple(offset)
+
+
+class ResizingOperator(ResizingOperatorBase):
+
+    """Operator mapping a discrete function to a new domain.
+
+    This operator is a mapping between uniformly discretized spaces with
+    the same `DiscreteLp.cell_sides` but different `DiscreteLp.shape`.
+    The underlying operation is array resizing, i.e. no resampling is
+    performed.
+    In axes where the domain is enlarged, the new entries are filled
+    ("padded") according to a provided parameter ``pad_mode``.
+
+    All resizing operator variants are linear, except constant padding
+    with constant != 0.
+
+    See `the online documentation
+    <https://odl.readthedocs.io/math/resizing_ops.html>`_
+    on resizing operators for mathematical details.
+    """
```

Comment:
This class is totally missing any examples, would be great with some simple examples.